subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/listener/ArtifactsTestListener.kt

```diff
@@ -67,26 +68,33 @@ internal class ArtifactsTestListener(
                     )
                 }
 
-            lifecycleListener.finished(
-                artifacts = resultDirectory,
-                test = test,
-                executionNumber = executionNumber
-            )
-
             device.clearDirectory(
                 remotePath = testMetadataDirectory.toPath()
             ).invoke()
+
+            resultDirectory
         } catch (t: Throwable) {
-            // TODO LOST
             logger.warn("Failed to process artifacts from $device", t)
+            Try.Failure(t)
         } finally {
             tempDirectory.toFile().delete()
         }
+
+        lifecycleListener.finished(
+            artifacts = artifacts,
+            test = test,
+            executionNumber = executionNumber
+        )
     }
 
     @Suppress("SdCardPath") // android API's are unavailable here
     private fun testMetadataFullDirectory(targetPackage: String, test: TestCase): File =
-        File("/sdcard/Android/data/$targetPackage/files/runner/${testMetadataDirectoryPath(test)}")
+        File("/sdcard/Android/data/$targetPackage/files/$RUNNER_OUTPUT_FOLDER/${testMetadataDirectoryPath(test)}")
 
     private fun testMetadataDirectoryPath(test: TestCase): String = "${test.className}#${test.methodName}"
+
+    companion object {
+        // todo should be passed with instrumentation params, see [ExternalStorageTransport]
```

Comment:
Why does anyone need to customize this path?

Reply:
it doesn't, but it's not clear if is it safe to change
you mean comment could be enough here?

Reply:
I see two options:
- Use value from constants. No one needs to customize it
- Client needs to customize the path

The question was about intention itself "should be passed with instrumentation params". 
Why we decided that this value should be customizable?

Reply:
yep, i agree
these two modules don't have a dependency from each other, so such constants need to be placed somewhere, and that's the right way;
i'll try to find such place and anyway rephrase the comment